scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+package org.oppia.android.scripts
+
+import java.io.File
+import java.io.FileInputStream
+import org.oppia.android.app.model.FilenameChecks
+import org.oppia.android.app.model.FilenameCheck
+import org.oppia.android.app.model.FileContentChecks
+import org.oppia.android.app.model.FileContentCheck
+import org.oppia.android.scripts.ScriptResultConstants
+
+class RegexPatternValidationCheck {
+  companion object {
+    @JvmStatic
+    fun main(vararg args: String) {
+
+      val repoPath = args[0] + "/"
+
+      val allowedDirectories: MutableList<String> = ArrayList()
+
+      for (layerIndex in 1 until args.size) {
+        allowedDirectories.add(args[layerIndex])
+      }
+
+      val searchFiles = collectSearchFiles(repoPath, allowedDirectories)
+
+      var scriptFailedFlag = false
+
+      getFilenameChecks().forEach {
+        if (checkProhibitedFileNamePattern(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            prohibitedFilenameRegexString = it.getProhibitedFilenameRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      getFileContentChecks().forEach {
+        if (checkProhibitedContent(
+            repoPath = repoPath,
+            searchFiles = searchFiles,
+            fileNameRegexString = it.getFilenameRegex(),
+            prohibitedContentRegexString = it.getProhibitedContentRegex(),
+            errorToShow = it.getFailureMessage()
+          )) {
+          scriptFailedFlag = true
+        }
+      }
+
+      if (scriptFailedFlag) {
+        throw Exception(ScriptResultConstants.REGEX_CHECKS_FAILED)
+      } else {
+        println(ScriptResultConstants.REGEX_CHECKS_PASSED)
+      }
+    }
+
+    /**
+     * Fetches all filename checks
+     *
+     * @return [List<FilenameCheck>] a list of all the FilenameChecks
+     */
+    fun getFilenameChecks(): List<FilenameCheck> {
+      val fileNamePatternsBinaryFile =
+        File("scripts/assets/filename_pattern_validation_checks.pb")
+      val filenameCheckBuilder = FilenameChecks.newBuilder()
+      val namePatternsObj: FilenameChecks =
+        FileInputStream(fileNamePatternsBinaryFile).use {
+          filenameCheckBuilder.mergeFrom(it)
+        }.build() as FilenameChecks
```

Comment:
I suggest factoring this & the common code below into a private helper to be shared by both.

Something that might help: https://github.com/oppia/oppia-android/blob/f9eb3cdcba174fdd78690af33ffe3c7a8b547a10/utility/src/main/java/org/oppia/android/util/extensions/BundleExtensions.kt#L32.

Reply:
Thanks, done.